subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/resolver/DefaultGradleApiSourcesResolver.java

```diff
@@ -148,12 +148,21 @@ private Dependency gradleSourceDependency() {
         Map<String, String> sourceDependency = new HashMap<>();
         sourceDependency.put("group", "gradle");
         sourceDependency.put("name", "gradle");
-        sourceDependency.put("version", gradleVersion().getVersion());
+        sourceDependency.put("version", dependencyVersion());
         sourceDependency.put("classifier", "src");
         sourceDependency.put("ext", "zip");
         return project.getDependencies().create(sourceDependency);
     }
 
+    private String dependencyVersion() {
+        GradleVersion version = gradleVersion();
+        if (version.isSnapshot()) {
+            return String.format("(6.1.1, %s]", version.getVersion());
```

Comment:
I am not happy about putting a hardcoded version here.
I put the range here in the first place because we do not have the _current_ under test snapshot sources in the repo.
Kotlin's implementation tries to compute the previous version to put on the lower bound on the fly. I could add that here as well, but on the other hand I thought why not keep it simpler?
What we really want here is - if the snapshot sources are not in repo, pick the previous latest snapshot.

Originally, the implementation would just skip the sources if they are not available, which I think is even better. However, in order to replicate the issue in a test, we want to download something because Kotlin's side does it this way. If we don't download, then transforms don't kick in and there is no conflict.

I spent quite a bit of time today trying to instead inject the `GRADLE_VERSION_OVERRIDE` env var to tooling API that would override the version for the test, but had no success.

Any better ideas to solve this?

Reply:
Could we stage the built distributions in a way that we could substitute GRADLE_REPO_URL for that local file path?